documentation/database.md

```diff
@@ -0,0 +1,296 @@
+# Database
+
+> [!WARNING]
+> This documentation is meant to describe the state of the database. The reader should be aware that some undocumented changes may have been done after its last updates, and for this purpose should check the git history of this file.
+>
+> Also contributions are welcome! :heart:
+
+First of all, let's explicit two databases that Grist manages:
+1. The Home Database;
+2. The Document Database (aka the grist document);
+
+The Home database is responsible for things related to the instance, like:
+ - the users and the groups registered on the instance;
+ - the billing;
+ - the organisations (aka sites), the workspaces;
+ - the documents metadata (id, name, workspace under which it is located...);
+ - the rights (ACL) to access to organisations, workspaces and documents (the access to the content of the document is controlled by the document itself);
+
+A Grist Document is a Sqlite database which contains data like:
+ - The tables, pages, views data;
+ - The ACL *inside* to access to all or part of tables (rows or columns);
+
+## The Document Database
+
+### Inspecting the Document
+
+A Grist Document (with the `.grist` extension) is actually a sqlite database. You may download a document like [this one](https://api.getgrist.com/o/templates/api/docs/keLK5sVeyfPkxyaXqijz2x/download?template=false&nohistory=false) and inspect its content using the `sqlite3` command:
+
+````
+$ sqlite3 Flashcards.grist
+sqlite> .tables
+Flashcards_Data                   _grist_TabBar
+Flashcards_Data_summary_Card_Set  _grist_TabItems
+GristDocTour                      _grist_TableViews
+_grist_ACLMemberships             _grist_Tables
+_grist_ACLPrincipals              _grist_Tables_column
+_grist_ACLResources               _grist_Triggers
+_grist_ACLRules                   _grist_Validations
+_grist_Attachments                _grist_Views
+_grist_Cells                      _grist_Views_section
+_grist_DocInfo                    _grist_Views_section_field
+_grist_External_database          _gristsys_Action
+_grist_External_table             _gristsys_ActionHistory
+_grist_Filters                    _gristsys_ActionHistoryBranch
+_grist_Imports                    _gristsys_Action_step
+_grist_Pages                      _gristsys_FileInfo
+_grist_REPL_Hist                  _gristsys_Files
+_grist_Shares                     _gristsys_PluginData
+````
+
+:warning: Be sure to work on a copy of a document if you inspect its content, otherwise you may loose data.
+
+### The migrations
+
+The migrations are handled in the python sandbox in this code:
+https://github.com/gristlabs/grist-core/blob/main/sandbox/grist/migrations.py
+
+For more information, please consult [the documentation for migrations](./migrations.md).
```

Comment:
We can add information of `sandbox/grist/schema.py` that are the schema of a Doc Database and we create a new migration we have to adapt this file

Reply:
The link to [./migrations.md](https://github.com/gristlabs/grist-core/blob/main/documentation/migrations.md) gives a indirect reference to this file with more context. What do you think?